### ISSUE TITLE
Incorrect Keyboard Focus Order in "AIGC-Test Extension" Window: A11y_Visual Studio Code_AIGC Window_FocusOrder

### ISSUE THREAD

**Issue by ManishaC1** [microsoft/vscode#201296]
[[Check out Accessibility Insights!](https://accessibilityinsights.io/)]- Identify accessibility bugs before check-in and make bug fixing faster and easier.

## GitHubTags:
#A11yTCS; #A11ySev2; #DesktopApp; #Win32; #Visual Studio Code Client; #BM-VisualStudioCodeClient-Win32-Aug2023; #WCAG2.4.3; #FocusOrder; #AILimited; #WWL:Bug1643902; #A11yMAS;

## Environment and OS details:

Version: 1.85.0-insider (user setup)
Commit: b3a649fe52c0fb8692c9b28dfe25227c97521884
Date: 2023-11-15T09:43:13.091Z
Electron: 25.9.4
ElectronBuildId: 25127168
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.22621

Windows 11 22H2(OS Build 22621.2821)

## Pre-requisites:

Here are the steps for using the AIGC extension in VS Code from scratch:
1. Download and install the latest version of VS Code from the official website: [https://code.visualstudio.com/download](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2f%2fcode.visualstudio.com%2fdownload&data=05%7c01%7cv-obulreddyv%40microsoft.com%7c19d1a2195b2b4507b83808db689e292b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c638218800757080404%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c3000%7c%7c%7c&sdata=8oeOMElEe0o2mdDGKtxteIulMqGzpV2hcIP6JYG4ScE%3D&reserved=0).

- Choose the right version to match your OS and select the standalone installer.

- Once downloaded, double click the installer to install VS Code with default settings.

2. Download and install Git from the official website: [https://git-scm.com/downloads](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2f%2fgit-scm.com%2fdownloads&data=05%7c01%7cv-obulreddyv%40microsoft.com%7c19d1a2195b2b4507b83808db689e292b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c638218800757080404%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c3000%7c%7c%7c&sdata=x%2B3zxf4llQSsZOM0U7gck1Wb993UvFZeEsXl6wBNQX8%3D&reserved=0).

- Choose the right version to match your OS and select the default settings during installation.

3. Open VS Code by searching for it in the Windows start menu.

4. Clone the repository from Azure DevOps .

- Switch to Source Control, click on Clone repository, paste the Repo URL([https://microsoftdigitallearning.visualstudio.com/DefaultCollection/Courseware/_git/LP_Sample_Qijie-Test](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2f%2fmicrosoftdigitallearning.visualstudio.com%2fDefaultCollection%2fCourseware%2f_git%2fLP_Sample_Qijie-Test&data=05%7c01%7cv-obulreddyv%40microsoft.com%7c19d1a2195b2b4507b83808db689e292b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c638218800757080404%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c3000%7c%7c%7c&sdata=hOLeTQfTI8p%2fHQ%2B%2Bi5swZk5zm68rzHNwVlzNnicX%2fys%3D&reserved=0)), choose a local folder to store the content, and wait for the cloning process to complete.

- Click Open in New Window and select "Yes, I trust the authors" in the pop-up window.

5. Install the AIGC extension for VS Code.

- Download the AIGC extension file ([ AIGC-Test-0.0.46.vsix](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2f%2fmicrosoft.sharepoint.com%2f%3Au%3A%2ft%2fLeXPDETS%2fEaO6BJywKChDiDdgFmqvVPEBiRWSI1vZU_kPk9P9L94cqw%3Fe%3Di6c0J1&data=05%7c01%7cv-obulreddyv%40microsoft.com%7c19d1a2195b2b4507b83808db689e292b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c638218800757080404%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c3000%7c%7c%7c&sdata=qN3hNayhcC73sCoVRWq4ZDO84aG3jkvX3StecaxaaJ4%3D&reserved=0)), select Extensions in VS Code, click three dots, select "Install from VSIX...", navigate to the extension file, select it, and click Install.

- The AIGC-Test icon should appear in the Installed extension list.

6. Click on the AIGC extension icon in the left side of VS Code to open the AIGC window.

- To move it to the right side, select View, click on Appearance, select "Secondary Side Bar", and drag the AIGC icon to the right side.

7. Open a markdown file under the includes folder, select the entire text, right click, and select "AIGC: Generate Knowledge check".

- In the AIGC window, verify the prompt is automatically selected as "Design knowledge check questions on <selected content>", type a number in "Number of Questions", and click the Send icon.

- Wait for the AIGC response with two knowledge check questions.

- To insert the generated content to your file, put the cursor at the end of the content, click Insert in the AIGC window, and verify that the two questions are appended to the content.

## Repro Steps:

1. Launch Visual Studio
2. Install AIGC-Test Extension.
3. Tab within the AIGC-Test Extension section, check whether focus order is logical.

 ## Actual Result:

Keyboard Focus Order within "AIGC-Test Extension" window is not correct. When the focus is on the "Close" button and the enter key is pressed, the focus shifts to the "Status bar" located outside of the extension window. And with the next Tab key is pressed, the focus moves onto the prompt within the extension.

## Expected Result:

Upon pressing the enter key while the focus is on the "Close" button, the keyboard focus should remain within the confines of the extension window.
Close button-->Clear chat button-->Insert button-->Edit section

## Attachment:
[Visual Studio Code_Incorrect focus Order.zip](https://github.com/microsoft/vscode/files/13727475/Visual.Studio.Code_Incorrect.focus.Order.zip)

**Comment by ManishaC1** [comment]
WWL Bug link-->https://dev.azure.com/worldwidelearning/WWL%20Accessibility/_workitems/edit/1643902

**Comment by meganrogge** [comment]
I am not sure what you mean. Can you please provide a gif?

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!

**Comment by ManishaC1** [comment]
@meganrogge 
Please activate the bug.

I have attached a gif capturing incorrect focus order using NVDA screen reader. Without screen reader it would be hard to understand about the whereabouts of the focus order. Additionally, I've included another detailed video addressing the issue.
Observed focus order: close button-->status bar present outside of chat window-->edit section present within chat section.
Expected focus order: close button-->clear chat button-->Insert button-->edit section.

![Presentation1](https://github.com/microsoft/vscode/assets/105198219/3119d1a6-310b-4e9c-a863-d0bb55ec8544)


https://github.com/microsoft/vscode/assets/105198219/4cbc8898-75aa-4419-9aa0-6e5b0f09cda1

**Comment by meganrogge** [comment]
thanks for the gifs. Is there any way to access the `clear chat` and `insert` actions once you're within the side panel?

**Comment by meganrogge** [comment]
when I follow your repro steps, I get an error so cannot continue. Furthermore, this seems like a niche edge case. Could you pls provide repro steps that demonstrate a more general/ easier to hit issue? Thanks.

Also cc @isidorn in case i'm misunderstanding

![Image](https://github.com/microsoft/vscode/assets/29464607/60ec0cd3-51c6-4542-9c87-9a2a0443493a)
